rskj-core/src/main/java/org/ethereum/vm/VM.java

```diff
@@ -1440,17 +1441,18 @@ protected void doCREATE(){
             throw Program.ExceptionHelper.modificationException(program);
         }
 
-        DataWord size;
+        DataWord codeSize;
         long sizeLong;
         long newMemSize ;
 
         if (computeGas) {
             gasCost = GasCost.CREATE;
-            size = stack.get(stack.size() - 3);
-            sizeLong = Program.limitToMaxLong(size);
+            codeSize = getCodeSize();
+            sizeLong = Program.limitToMaxLong(codeSize);
             checkSizeArgument(sizeLong);
             newMemSize = memNeeded(stack.get(stack.size() - 2), sizeLong);
-            gasCost = GasCost.add(gasCost, calcMemGas(oldMemSize, newMemSize, 0));
+            gasCost = GasCost.add(gasCost, calcMemGas(oldMemSize, newMemSize, 0))
+                    + InitcodeCostCalculator.getInstance().calculateCost(codeSize.intValue(), program.getActivations());
```

Comment:
do you think we could make use of the `GasCost.add()` helper for adding initcode cost?

Reply:
Ok, will do.